subprojects/workers/src/main/java/org/gradle/workers/internal/ActionExecutionSpec.java

```diff
@@ -16,9 +16,16 @@
 
 package org.gradle.workers.internal;
 
+import org.gradle.internal.reflect.Instantiator;
+
+import javax.annotation.Nullable;
+
 public interface ActionExecutionSpec extends WorkSpec {
     Class<? extends Runnable> getImplementationClass();
 
+    @Nullable
+    Instantiator getActionInstantiator();
```

Comment:
I know we talked about this when we chatted the other day, but now that I look at it, I don't think this is the way to go.  I don't think this will ever work with the isolated classloader case since it serializes the spec on its way.  I think it should probably just be specified with the invocation.